### PR TITLE
feat: V6 updates to toggle, library, setup, analytics and storage

### PR DESCRIPTION
**Pull request by stocaaro**
#### Description of changes:
Putting out an PR against the V6 branch updates.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [x] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [x] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib-v1/storage/js/configureaccess.mdx
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/configureaccess.mdx:41]:
```suggestion
You can customize your key path by defining a prefix resolver:

```typescript
Amplify.configure(
    {},
    {
        Storage: {
            S3: {
                prefixResolver: async ({ accessLevel, targetIdentityId }) => {
                    if (accessLevel === 'guest') {
                        return 'myPublicPrefix/';
                    } else if (accessLevel === 'protected') {
                        return `myProtectedPrefix/${targetIdentityId}`;
                    } else {
                        return `myPrivatePrefix/${targetIdentityId}`;
                    }
                },
            },
        },
    }
);
```
```

File: src/fragments/lib-v1/storage/js/download.mdx
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/download.mdx:20]: This indentiaon seems to be off

File: src/fragments/lib-v1/storage/js/download.mdx
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/download.mdx:48]: Ditto indentation

File: src/fragments/lib-v1/storage/js/download.mdx
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/download.mdx:56]:
```suggestion
await getUrl('filename.txt', { expiresIn: 60 });
```

File: src/fragments/lib-v1/storage/js/download.mdx
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/download.mdx:18]: Looks good for now, but I'd like to add a advanced use case section later showing how to get a long-live(7 days) presigned URL

File: src/fragments/lib-v1/storage/js/download.mdx
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/download.mdx:84]: Ditto indentation

File: src/fragments/lib-v1/storage/js/download.mdx
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/download.mdx:92]:
```suggestion
    console.log(`Downloaded: ${progress.transferredBytes}/${progress.totalBytes}`);
```

File: src/fragments/lib-v1/storage/js/existing-resources.mdx
Comment (AllanZhengYP) [inline src/fragments/lib-v1/storage/js/existing-resources.mdx:20]:
```suggestion
Amplify.configure({
    Storage: {
        S3: {
            region: 'us-east-1', // (required) - Amazon S3 bucket region
            bucket: 'testappa321fb103f1f2ae6a4d25d8cd2161728123152-dev' // (required) - Amazon S3 bucket URI
        }
    }
});
```

File: src/fragments/lib-v1/analytics/js/identifyuser.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/identifyuser.mdx:3]:
```suggestion
This API sends information about the current user to Amazon Pinpoint.

Additional information such as the user's name, email, location, and device can be included by specifying the `UserProfile`. Custom attributes can also be included by setting `UserProfile.attributes`.

You can provide `name`, `email` and `plan`, as well as location information with `AnalyticsUserProfile.Location`. You can also send additional custom attributes using  `AnalyticsProperties`.
```

File: src/fragments/lib-v1/analytics/js/identifyuser.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/identifyuser.mdx:5]:
```suggestion
If the user was signed in through [signIn](/lib/auth/signin) you can retrieve the current user's ID as shown below:
```

File: src/fragments/lib-v1/analytics/js/identifyuser.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/identifyuser.mdx:38]:
```suggestion
async function sendUserData() {
    const user = await getCurrentUser(); 

    identifyUser({
        userId: user.userId,
        userProfile: userProfile
    });
}
```

File: src/fragments/lib-v1/analytics/js/identifyuser.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/identifyuser.mdx:12]: Await can't be used at module-scope

```suggestion
```

File: src/fragments/lib-v1/analytics/js/identifyuser.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/identifyuser.mdx:22]:
```suggestion
const attributes = {
        name: "username",
        email: "name@example.com",
        plan: "plan",
```

File: src/fragments/lib-v1/analytics/js/identifyuser.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/identifyuser.mdx:30]:
```suggestion
```

File: src/fragments/lib-v1/analytics/js/record.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/record.mdx:13]:
```suggestion
The `record` API lets you add additional attributes to an event. For example, to record _artist_ information with an _albumVisit_ event:
```

File: src/fragments/lib-v1/analytics/js/record.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/record.mdx:22]:
```suggestion
record({
    event: {
        name: 'albumVisit',
        attributes: { genre: '', artist: '' }
    }
});
```

File: src/fragments/lib-v1/analytics/js/record.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/record.mdx:27]:
```suggestion
Metrics can also be added to an event:
```

File: src/fragments/lib-v1/analytics/js/record.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/record.mdx:36]:
```suggestion
record({
    record: {
        name: 'albumVisit',
        metrics: { minutesListened: 30 }
    }
});
```

File: src/fragments/lib-v1/analytics/js/record.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/record.mdx:39]:
```suggestion
Metric values must be a `number` type such as a float or integer.
```

File: src/fragments/lib-v1/analytics/js/record.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/analytics/js/record.mdx:52]: This feature is not supported until GA:

```suggestion
```

File: src/fragments/lib-v1/start/getting-started/reactnative/setup.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/start/getting-started/reactnative/setup.mdx:1]: Should remove the React Native getting started guide as it's not supported for Dev Preview

File: src/fragments/lib-v1/start/getting-started/vanillajs/setup.mdx
Comment (jimblanc) [inline src/fragments/lib-v1/start/getting-started/vanillajs/setup.mdx:101]: We'll need to re-work this as DataStore won't be available in preview. Okay doing it in a follow-up PR

File: src/pages/lib-v1/q/platform/[platform].mdx
Comment (jimblanc) [inline src/pages/lib-v1/q/platform/[platform].mdx:41]:
```suggestion
Welcome to the developer preview of **Amplify JavaScript v6**. This preview introduces functional interfaces, typescript improvements and improved NextJS SSR support  for Authentication, Analytics, and Storage.
```

File: src/pages/lib-v1/q/platform/[platform].mdx
Comment (jimblanc) [inline src/pages/lib-v1/q/platform/[platform].mdx:43]:
```suggestion
The team is working on the other categories for the v6 release. This developer preview is offered for you to try it out and provide feedback on the coming changes. 
```

**Review by jimblanc — COMMENTED** [review]
Thank you so much for doing this! Dropped some initial feedback specifically on Analytics if we could merge in this PR, but this is a great base to start iterating on for more discrete changes.

**Review by jimblanc — APPROVED** [review]
Approving to unblock incremental changes & updates

File: src/pages/lib-v1/q/platform/[platform].mdx
Comment (ashwinkumar6) [inline src/pages/lib-v1/q/platform/[platform].mdx:45]: I'm wondering if we should we comment this line

Comment (stocaaro) [reply src/pages/lib-v1/q/platform/[platform].mdx:45]: Leaving it in so that it gets further attention before we merge it.

**Pull request merged**
